amostra/server/engine.py

```diff
@@ -338,13 +338,13 @@ def post(self):
             uids.append(data['uid'])
             res = database.container.insert(data)
             if res:
-                database.request.create_index([('uid', pymongo.DESCENDING)],
-                                                 unique=True, background=True)
-                database.request.create_index([('time', pymongo.DESCENDING)],
-                                            unique=False, background=True)
+                database.container.create_index([('uid', pymongo.DESCENDING)],
```

Comment:
The connection pool doesn't close as soon as you insert. Driver parses the indexing efficiently so I wouldn't worry about network hits. Since we don't have the latest mongo backend in all beamlines and old versions recommend ensure_index, I am hesitant to change this as I prefer robustness over a few micro seconds of performance for single and bulk inserts.
My end goal is to do the indexing in a smart way using jsonschema as soon as mongo backends are up to date. Let's talk offline